components/brave_ads/core/internal/common/strings/string_conversions_util.cc

```diff
@@ -18,4 +22,30 @@ std::string BoolToString(const bool value) {
   return value ? kTrue : kFalse;
 }
 
+std::vector<float> ConvertStringToVector(const std::string& string,
+                                         const std::string& delimiter) {
+  const std::vector<std::string> vector_string = base::SplitString(
+      string, delimiter, base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  std::vector<float> vector;
+  vector.reserve(vector_string.size());
```

Comment:
`std::vector<float> list(components.size()`

Reply:
if I pre-initialize instead of using `reserve` then I cannot use `push_back` and the general iterator